config/image.go

```diff
@@ -73,15 +76,19 @@ func (c *ImageConfig) Validate(path pth.Path, config *Config) *pth.Error {
 }
 
 func (c *ImageConfig) validateBuildOrPull() error {
-	if c.Dockerfile == "" && c.Context == "" && !c.Pull.IsSet() {
-		return fmt.Errorf("one of dockerfile, context, or pull is required")
+	if len(c.Dobifile) == 0 && c.Dockerfile == "" && c.Context == "" && !c.Pull.IsSet() {
+		return fmt.Errorf("one of dockerfile, dobifile, context, or pull is required")
+	}
+	if c.Dockerfile != "" && len(c.Dobifile) != 0 {
+		return fmt.Errorf("Either use a dockerfile or a dobifile")
```

Comment:
This isn't right, they could use `pull:` without either of these, and that would be valid.